api/src/main/java/jakarta/servlet/ServletInputStream.java

```diff
@@ -50,6 +52,63 @@ public abstract class ServletInputStream extends InputStream {
     protected ServletInputStream() {
     }
 
+    /**
+     * Reads from the input stream into the given buffer. Returns the number of bytes read or {@code -1} if the end of
+     * stream is reached without reading any data.
+     * <p>
+     * If this method is called when {@code buffer} has no space remaining, the method returns {@code 0} immediately.
```

Comment:
What do you think about making this an error instead? The current implementation might make it a bit easier for buggy application code to get stuck in an infinite loop.

Reply:
Returning zero is consistent with `InputStream.read(byte[])` and with `SocketChannel.read(ByteBuffer)`. I understand the motivation but my preference is for consistency.

Reply:
Does this immediate return of 0 happen even if the method is called in non blocking mode with isReady false?
I think that should still be an error. Maybe leave out "immediately"?

Reply:
I agree that should still be an ISE and that is what the provided implementation does. I'll see what I can do to clarify the wording.